src/sentry/static/sentry/app/components/platformPicker.tsx

```diff
@@ -13,28 +12,34 @@ import EmptyMessage from 'app/views/settings/components/emptyMessage';
 import ExternalLink from 'app/components/links/externalLink';
 import ListLink from 'app/components/links/listLink';
 import NavTabs from 'app/components/navTabs';
-import categoryList from 'app/data/platformCategories';
+import categoryList, {PlatformKey} from 'app/data/platformCategories';
 import platforms from 'app/data/platforms';
 import space from 'app/styles/space';
 import {IconClose, IconSearch, IconProject} from 'app/icons';
+import {PlatformIntegration} from 'app/types';
 
-const PLATFORM_CATEGORIES = categoryList.concat({id: 'all', name: t('All')});
+const PLATFORM_CATEGORIES = [...categoryList, {id: 'all', name: t('All')}] as const;
 
-class PlatformPicker extends React.Component {
-  static propTypes = {
-    setPlatform: PropTypes.func.isRequired,
-    platform: PropTypes.string,
-    showOther: PropTypes.bool,
-    listClassName: PropTypes.string,
-    listProps: PropTypes.object,
-    noAutoFilter: PropTypes.bool,
-  };
+type Props = {
+  setPlatform: (key: PlatformKey | null) => void;
+  platform?: string | null;
```

Comment:
So this can be a string, `null` or `undefined`? Is it not possible to tighten this up a bit more?

Reply:
Yeah there's a few use cases here:

1. You can use this without indicating a selected platform, and just using it to select a platform.
2. You can pass `null` which indicates that you cleared a previously selected paltform.
3. And the platform string (ideally this would be `PlatformKey` but there's some reason it couldn't be that I didn't want to fix)